### ISSUE TITLE
Typer 0.12.2 breaks test suite

### ISSUE THREAD

**Issue by aaazzam** [PrefectHQ/prefect#12617]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

[Typer 0.12.2](https://github.com/tiangolo/typer/releases/tag/0.12.2) breaks OSS test suite, specifically `test_pluralized_subcommands_have_multiple_valid_invocations` in our cli tests. 

Specifically https://github.com/tiangolo/typer/pull/567, which appears to build a list of subcommands for better rendering but in doing so calls max instead of length on an empty array, which throws an error. 

### Reproduction

```python3
run the test suite on 0.12.2
```

### Versions

```Text
Version:             2.16.9+17.g3feaafd00.dirty
API version:         0.8.4
Python version:      3.11.8
Git commit:          3feaafd0
Built:               Mon, Apr 8, 2024 1:17 PM
OS/Arch:             darwin/arm64
Profile:             staging
Server type:         cloud
```

**Comment by svlandeg** [comment]
Sorry about this! I think https://github.com/tiangolo/typer/pull/796 will fix things again.

**Comment by tiangolo** [comment]
Typer 0.12.3 just released :rocket: 

Thanks @svlandeg for the fix!

**Comment by svlandeg** [comment]
> just need to figure out whether I need to exclude it in perpetuity or not

It's up to you, of course, but if this bug was only breaking your test suite I think the new release should fix that for you, as the test suite will always install the latest. Other than that, I'd expect this bug to show up only in edge cases, so I'm not sure it's worth yanking for.

**Comment by zhen0** [comment]
@aaazzam - since this is fixed in Typer 0.12.3, I'm closing this issue.  Feel free to let me know if you want it re-opened.
